### PR TITLE
fea: issue 2 add item and list view routes built plus navigation links

### LINKED ISSUE #2: 2. As a user, I want to be able to navigate between the “list” view and the “add an item” view so that the interface is clearly delineated between viewing and adding.
This is 1 of 2 stories that should be completed in the first week to enable further development of the app.

      AC:
      * `react-router-dom` has been added as a project dependency
      * Links are present and persistent at the bottom of the app: one for the “list” view, the other for the “add an item” view
      * When one of the links is clicked, the browser URL updates to represent the current view and a matching view component is displayed
      * Whichever view is selected, the corresponding link should display in bold text
      
      Notes:
      * Some pairs have gone straight to wanting to integrate a CSS framework to achieve the “links pinned to the bottom of the page” AC here. You're encouraged to keep it simple at this point. Perhaps look into `position: fixed;` or perhaps look into a tutorial on CSS grid. There will be time to get fancy later in the project.

### PR DESCRIPTION
**Pull request by alenamedved**
_For an example of how to fill this template out, [see this Pull Request](https://github.com/the-collab-lab/tcl-3-smart-shopping-list/pull/44)._ 

## Description

* React-router-dom has been added.
**Note: React-router-dom was updated from v5 to v6. The changes that affected our project:** 
1. Switch Replaced With Routes
2. Route takes a prop called 'element' where we have to pass a JSX component for that to be rendered
3. 'exact' prop is not needed anymore
4. NavLink accepts a function to control the styles which in turn gives up some information about the link (isActive)
follow the link to find more info about v6 [link](https://dev.to/arunavamodak/react-router-v5-vs-v6-dp0)

* Links are present and persistent at the bottom of the app: one for the “list” view, the other for the “add an item” view
* When one of the links is clicked, the browser URL updates to represent the current view and a matching view component is displayed
* Whichever view is selected, the corresponding link should display in bold text
  
## Related Issue
closes #2 

## Acceptance Criteria
  * `react-router-dom` has been added as a project dependency
  * Links are present and persistent at the bottom of the app: one for the “list” view, the other for the “add an item” view
  * When one of the links is clicked, the browser URL updates to represent the current view and a matching view component is displayed
  * Whichever view is selected, the corresponding link should display in bold text

## Type of Changes

|     | Type                       |
| --- | -------------------------- |
|    | :bug: Bug fix              |
| ✓  | :sparkles: New feature     |
|    | :hammer: Refactoring       |
|    | :100: Add tests            |
|  ✓  | :link: Update dependencies |
|    | :scroll: Docs              |

### Before
before

### After
after

AddItemPage

ListView

## Testing Steps / QA Criteria
run npm install or npm i to add react-router to dependencies

### CODE REVIEW

**Comment by rudidev08** [comment]
@alenamedved @shelleymcq I love that react router v6 upgrade was noted in comments. Since it's significant change, I'd have that as a top level bullet point with something like "using react router v6 instead of default vX. Notable change: (insert whatever seems to be huge change, if any, that you'd want coworkers to know). See https://dev.to/arunavamodak/react-router-v5-vs-v6-dp0 for more details". I'd even start that bullet item with "Note" that's bold or different color to draw attention, since it's significant change compared to other projects and it's not missed.

Current code (IMHO) is clean and obvious and doesn't need comments; for future definitely keep thinking as you're coding if you in 6 months from now, or with wiped memory, or coworker who has never seen the code could use comments. App will start to get more complex soon.

**Review by rudidev08 — APPROVED** [review]

**Comment by djtaylor8** [comment]
Looks good! Simple and to the point - nice! I wonder if at some point we would want to circle back and change the names of the routes? A small thing, but maybe something to consider.

**Comment by djtaylor8** [comment]
Looks good! Simple and to the point - nice! I wonder if at some point we would want to circle back and change the names of the routes? A small thing, but maybe something to consider.

**Comment by djtaylor8** [comment]
Looks good! Simple and to the point - nice! I wonder if at some point we would want to circle back and change the names of the routes? A small thing, but maybe something to consider.

**Review by djtaylor8 — COMMENTED** [review]
Looks good! Simple and to the point - nice! I wonder if at some point we would want to circle back and change the names of the routes? A small thing, but maybe something to consider.

File: src/App.js [hunk 1]
```diff
@@ -1,25 +1,23 @@
...
-          rel="noopener noreferrer"
-        >
-          Learn React
-        </a>
-      </header>
-    </div>
+    <>
+      <Routes>
+        <Route path="/" element={<Home />} />
+        <Route path="addItem" element={<AddItem />} />
```
Comment (djtaylor8) [inline src/App.js]: Maybe we can circle back and look into updating route names here

**Review by djtaylor8 — APPROVED** [review]

File: src/App.js (hunk 1 above)
Comment (alenamedved) [reply src/App.js]: sure, we can discuss the names. If you have any ideas post it here or we discuss tomorrow during our meeting?

**Review by lindseyindev — APPROVED** [review]
Sorry for the delay I had some technical issues on my end I have it working now and this looks great guys! Congrats on your first pull request here!

**Comment by lindseyindev** [comment]
We might need some help on this part.. I also had merge conflicts when pulling down from the package-lock.json and I think I used git pull --force to resolve it not sure how it would be resolved on github

conflicts

File: src/App.js (hunk 1 above)
Comment (alenamedved) [reply src/App.js]: I've changed addItem  path to addItemView

File: src/components/Navigation.jsx [hunk 2]
```diff
@@ -0,0 +1,22 @@
+import React from 'react';
+import { NavLink } from 'react-router-dom';
+
+export default function Navigation() {
+  return (
+    <nav>
```
Comment (laurenyz) [inline src/components/Navigation.jsx]: Great use of semantic HTML and ternaries!

**Review by laurenyz — COMMENTED** [review]
Wow - the bullet points explaining the changes in `react-router-dom v6` were great! Very easy to follow and understand!

The only thing from the ACs that I think could be improved is this part:

> Links are present and persistent at the bottom of the app

The navigation component lives underneath the other components, but my interpretation of the AC is that it stays towards the bottom of the page when the url changes. It could be interpreted other ways, but I think it makes sense that the nav bar would remain in a specific spot on the page even when the other components/pages develop and change sizes.

Very clear and clean PR - great job!

**Comment by alenamedved** [comment]
> Wow - the bullet points explaining the changes in `react-router-dom v6` were great! Very easy to follow and understand!
> 
> The only thing from the ACs that I think could be improved is this part:
> 
> > Links are present and persistent at the bottom of the app
> 
> The navigation component lives underneath the other components, but my interpretation of the AC is that it stays towards the bottom of the page when the url changes. It could be interpreted other ways, but I think it makes sense that the nav bar would remain in a specific spot on the page even when the other components/pages develop and change sizes.
> 
> Very clear and clean PR - great job!

Hi @laurenyz 
Thank you for the review. I agree with you, the part about navbar can be improved. I think we didn't consider it was important cause we were going to add css later and fix this imperfection. But for now to make it look good and follow AC 100% I've added css and made navbar  stick to the bottom.

**Comment by laurenyz** [comment]
Looks great, @alenamedved !

**Review by laurenyz — APPROVED** [review]

**Pull request merged**